### PR TITLE
expose interface to check CUDA support

### PR DESCRIPTION
**Pull request by simonbyrne**
cc: @kpamnany

### CODE REVIEW

**Comment by kpamnany** [comment]
LGTM.

Also, I manually tested this on Central with OpenMPI, with and without CUDA support, and with MPICH, without CUDA support. Works fine.

File: src/environment.jl [hunk 1]
```diff
@@ -158,3 +158,28 @@ function Wtime()
...
+Check if the MPI implementation is known to have CUDA support. Currently only Open MPI
+provides a mechanism to check, so it will return `false` with other implementations
+(unless overriden).
+
+This can be overriden by setting the `JULIA_MPI_HAS_CUDA` environment variable to `true`
+or `false`.
+"""
+function has_cuda()
+    flag = get(ENV, "JULIA_MPI_HAS_CUDA", nothing)
+    if flag === nothing
```
Comment (vchuravy) [inline src/environment.jl]: Could we use this to actually provide a function `MPI_vendor()`?

@leios wanted something like that.

Comment (simonbyrne) [reply src/environment.jl]: My plan is to do this as part of adding BinaryProvider support, but I don’t have the bandwidth to do this now

Comment (kpamnany) [reply src/environment.jl]: #354 does that but doesn't help with CUDA awareness.

Comment (simonbyrne) [reply src/environment.jl]: I've switched this to use #354.

**Pull request merged**